docs/guides/deployment-patterns.md

```diff
@@ -0,0 +1,170 @@
+
+This pattern is most useful for most common, straightforward flows. There’s only two levels of abstraction to think about—the flow itself and the tasks within it. It’s an easy pattern to set up, maintain, and use, especially if the flow is fully owned and maintained by a single person or team. However, this pattern can get unwieldy when there are multiple code owners, many tasks, or tasks with different infrastructure needs within the same flow.
+
+Here is an example of how a monoflow would look like:
+```python
+import httpx
+from prefect import flow, get_run_logger
+
+@flow
+def get_repo_info(repo_name: str = "PrefectHQ/prefect"):
+    url = f"https://api.github.com/repos/{repo_name}"
+    response = httpx.get(url)
+    response.raise_for_status()
+    repo = response.json()
+    logger = get_run_logger()
+    logger.info(f"PrefectHQ/prefect repository statistics 🤓:")
+    logger.info(f"Stars 🌠 : {repo['stargazers_count']}")
+    logger.info(f"Forks 🍴 : {repo['forks_count']}")
+
+if __name__ == "__main__":
+    get_repo_info()
+```
+
+### Flow of subflows
+![flow of deployments diagram](img/guides/subflow.png)
+
+Prefect’s orchestration engine is the first to offer first-class subflows. Any flow written with Prefect can be used as a component in another flow. A subflow has the same relationship to its parent flow as a task does. It runs in the same process as its parent flow. You can use subflows much as you would use an imported module in a python script.
+
+This pattern is most useful when you only want conceptual separation. It increases conceptual overhead in the form of additional layers of abstraction: the parent flow, its task and subflow runs, and the subflow runs’ tasks. In exchange, this pattern compartmentalizes the components of the parent process. For an individual, this can be useful for decomposing large flows into more easily reasoned logical units. For teams, it facilitates clear ownership boundaries and facilitates code-reuse.
+
+Lets try to configure the monoflow example to be a flow of subflows. 
+```python
+import httpx
+from prefect import flow
+
+@flow(log_prints = True)
+def get_repo_info():
+    url = 'https://api.github.com/repos/PrefectHQ/prefect'
+    api_response = httpx.get(url)
+    if api_response.status_code == 200:
+        repo_info = api_response.json()
+        stars = repo_info['stargazers_count']
+        forks = repo_info['forks_count']
+        contributors_url = repo_info['contributors_url']
+        average_commits = calculate_average_commits(contributors_url)
+        print(f"PrefectHQ/prefect repository statistics 🤓:")
+        print(f"Stars 🌠 : {stars}")
+        print(f"Forks 🍴 : {forks}")
+        print(f"Average commits per contributor 💌 : {average_commits:.2f}")
+    else:
+        raise Exception('Failed to fetch repository information.')
```

Comment:
we're using `raise_for_status` above, which I think makes more sense than raising a bare `Exception` if not 200.

```suggestion
    api_response.raise_for_status()
    repo_info = api_response.json()
    stars = repo_info['stargazers_count']
    forks = repo_info['forks_count']
    contributors_url = repo_info['contributors_url']
    average_commits = calculate_average_commits(contributors_url)
    print(f"PrefectHQ/prefect repository statistics 🤓:")
    print(f"Stars 🌠 : {stars}")
    print(f"Forks 🍴 : {forks}")
    print(f"Average commits per contributor 💌 : {average_commits:.2f}")
```

Reply:
Sounds good I can get those changes in!